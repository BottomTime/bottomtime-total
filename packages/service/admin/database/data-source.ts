import path from 'path';
import { DataSource } from 'typeorm';

export async function getDataSource(
  url: string,
  requireSsl: boolean,
): Promise<DataSource> {
  const client = new DataSource({
    type: 'postgres',
    url,
    entities: [path.resolve(__dirname, '../../src/data/**/*.entity.ts')],
    ssl: requireSsl,
  });
  return await client.initialize();
}
