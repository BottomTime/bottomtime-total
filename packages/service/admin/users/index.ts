import { CommandModule } from 'yargs';

import { createAdmin } from './create-admin';
import { getUserToken } from './get-user-token';

export const userModule: CommandModule<{
  'postgres-uri': string;
  'require-ssl': boolean;
}> = {
  command: 'user',

  describe: 'Commands for managing a user account',

  builder(yargs) {
    return yargs
      .command(
        'new-admin',
        'Creates a new admin user.',
        (yargs) => {
          return yargs
            .positional('username', {
              alias: 'u',
              description: "The user's username.",
              type: 'string',
            })
            .positional('password', {
              alias: 'p',
              description: 'The password to set on the new admin account',
              type: 'string',
            })
            .help();
        },
        async (yargs) => {
          await createAdmin({
            postgresRequireSsl: yargs.requireSsl,
            potgresUri: yargs.postgresUri,
            username: yargs.username,
            password: yargs.password,
          });
        },
      )
      .command(
        'token <user>',
        'Gets a token for a user',
        (yargs) => {
          return yargs
            .positional('user', {
              demandOption: true,
              description: "The user's username or email address.",
              type: 'string',
            })
            .help();
        },
        async (yargs) => {
          await getUserToken(yargs.postgresUri, yargs.requireSsl, yargs.user);
        },
      )
      .help();
  },

  async handler() {
    /* TODO */
  },
};
