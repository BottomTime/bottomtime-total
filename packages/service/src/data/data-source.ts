import path from 'path';
import { DataSource, DataSourceOptions } from 'typeorm';

import { Config } from '../config';

export const PostgresDataSourceOptions: DataSourceOptions = {
  type: 'postgres',
  url: Config.postgresUri,
  entities: [path.join(__dirname, './**/*.entity.ts')],
  synchronize: false,
};

export async function initDataSource(): Promise<DataSource> {
  const AppDataSource = new DataSource(PostgresDataSourceOptions);
  return await AppDataSource.initialize();
}
