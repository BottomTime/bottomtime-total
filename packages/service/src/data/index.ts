export * from './data-source';
export * from './get-query-vector';

export * from './certification.entity';
export * from './dive-site-review.entity';
export * from './dive-site.entity';
export * from './friend-request.entity';
export * from './friendship.entity';
export * from './tank.entity';
export * from './user-certification.entity';
export * from './user-oauth.entity';
export * from './user.entity';
