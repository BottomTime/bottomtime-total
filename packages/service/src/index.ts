import { Config } from './config';
import { createApp } from './create-app';
import { createLogger } from './logger';

const log = createLogger(Config.logLevel);

createApp(log)
  .then(async (app) => {
    await app.listen(Config.port);

    log.info(
      `ðŸŽ‰ Service has successfully started and is listening on port ${Config.port}. ðŸŽ‰`,
    );
  })
  .catch((error) => {
    log.fatal('[SERVICE] Failed to start service... Shutting down...', error);
    process.exit(1);
  });
