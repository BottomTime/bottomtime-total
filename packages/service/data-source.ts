import path from 'path';
import { DataSource } from 'typeorm';

import { PostgresDataSourceOptions } from './src/data/data-source';

export default new DataSource({
  ...PostgresDataSourceOptions,
  migrations: [path.join(__dirname, './migrations/*.ts')],
});
