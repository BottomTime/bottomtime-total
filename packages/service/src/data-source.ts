import 'reflect-metadata';
import { DataSourceOptions } from 'typeorm';

import { Config } from './config';
import * as Entities from './data';

export const PostgresDataSourceOptions: DataSourceOptions = {
  type: 'postgres',
  url: Config.postgresUri,
  entities: Object.values(Entities),
  synchronize: false,
};
