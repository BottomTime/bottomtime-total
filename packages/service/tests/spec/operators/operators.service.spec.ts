import { LogBookSharing, VerificationStatus } from '@bottomtime/api';

import { writeFile } from 'fs/promises';
import { resolve } from 'path';
import { Repository } from 'typeorm';

import { OperatorEntity, UserEntity } from '../../../src/data';
import {
  CreateOperatorOptions,
  OperatorsService,
} from '../../../src/operators/operators.service';
import { User } from '../../../src/users';
import { dataSource } from '../../data-source';
import TestData from '../../fixtures/operators.json';
import TestOwners from '../../fixtures/user-search-data.json';
import { createTestOperator, parseOperatorJSON } from '../../utils';
import { createTestUser, parseUserJSON } from '../../utils/create-test-user';

describe('OperatorService', () => {
  let Users: Repository<UserEntity>;
  let Operators: Repository<OperatorEntity>;
  let service: OperatorsService;

  let owner: User;

  beforeAll(() => {
    Users = dataSource.getRepository(UserEntity);
    Operators = dataSource.getRepository(OperatorEntity);

    service = new OperatorsService(Operators);
  });

  beforeEach(async () => {
    const userData = createTestUser({
      id: '54ca5e54-de92-4e15-a523-4087c52b40eb',
      username: 'testuser',
      memberSince: new Date('2021-06-10T03:00:00Z'),
      logBookSharing: LogBookSharing.Public,
      avatar: 'https://example.com/avatar.png',
      location: 'Toronto, ON, Canada',
      name: 'Test User',
    });
    owner = new User(Users, userData);
    await Users.save(userData);
  });

  it.skip('will make some sweet, sweet test data', async () => {
    const operators = new Array<OperatorEntity>(200);
    for (let i = 0; i < operators.length; i++) {
      operators[i] = createTestOperator();
    }

    await writeFile(
      resolve(__dirname, '../../fixtures/operators.json'),
      JSON.stringify(operators, null, 2),
      'utf-8',
    );
  });

  describe('when creating a new operator', () => {
    it('will create a new operator with minimal properties', async () => {
      const options: CreateOperatorOptions = {
        active: true,
        name: "Bill's Dive Shop",
        address: '123 Main St',
        description: 'The best dive shop in town!',
        slug: 'bills-dive-shop',
        owner,
      };

      const operator = await service.createOperator(options);

      expect(operator.id).toHaveLength(36);
      expect(operator.createdAt.valueOf()).toBeCloseTo(Date.now(), -3);
      expect(operator.updatedAt.valueOf()).toBeCloseTo(Date.now(), -3);
      expect(operator.name).toBe(options.name);
      expect(operator.owner.userId).toBe(owner.id);

      const saved = await Operators.findOneOrFail({
        where: { id: operator.id },
        relations: ['owner'],
      });
      expect(saved.createdAt).toEqual(operator.createdAt);
      expect(saved.updatedAt).toEqual(operator.updatedAt);
      expect(saved.name).toBe(operator.name);
      expect(saved.owner!.id).toBe(owner.id);
    });

    it('will create a new operator with full properties', async () => {
      const options: CreateOperatorOptions = {
        active: true,
        name: "Bill's Dive Shop",
        slug: 'bills-dive-shop',
        address: '123 Main St',
        description: 'The best dive shop in town!',
        email: 'shop@bill.edu',
        gps: {
          lat: 43.6532,
          lon: -79.3832,
        },
        phone: '123-456-7890',
        socials: {
          facebook: 'billsdive',
          instagram: 'billsdive',
          tiktok: '@billsdive',
          twitter: 'billsdive',
          youtube: 'billsdive',
        },
        owner,
        website: 'https://billsdive.com',
      };

      const operator = await service.createOperator(options);

      expect(operator.id).toHaveLength(36);
      expect(operator.createdAt.valueOf()).toBeCloseTo(Date.now(), -3);
      expect(operator.updatedAt.valueOf()).toBeCloseTo(Date.now(), -3);
      expect(operator.name).toBe(options.name);
      expect(operator.owner.userId).toBe(owner.id);

      expect(operator.address).toBe(options.address);
      expect(operator.description).toBe(options.description);
      expect(operator.email).toBe(options.email);
      expect(operator.gps).toEqual(options.gps);
      expect(operator.phone).toBe(options.phone);
      expect(operator.socials.toJSON()).toEqual(options.socials);
      expect(operator.website).toBe(options.website);

      const saved = await Operators.findOneOrFail({
        where: { id: operator.id },
        relations: ['owner'],
      });
      expect(saved.createdAt).toEqual(operator.createdAt);
      expect(saved.updatedAt).toEqual(operator.updatedAt);
      expect(saved.name).toBe(operator.name);
      expect(saved.owner!.id).toBe(owner.id);

      expect(saved.address).toBe(operator.address);
      expect(saved.description).toBe(operator.description);
      expect(saved.email).toBe(operator.email);
      expect(saved.gps).toEqual({
        type: 'Point',
        coordinates: [operator.gps!.lon, operator.gps!.lat],
      });
      expect(saved.phone).toBe(operator.phone);
      expect(saved.website).toBe(operator.website);
      expect(saved.facebook).toBe(operator.socials.facebook);
      expect(saved.instagram).toBe(operator.socials.instagram);
      expect(saved.tiktok).toBe(operator.socials.tiktok);
      expect(saved.twitter).toBe(operator.socials.twitter);
      expect(saved.youtube).toBe(operator.socials.youtube);
      expect(saved.website).toBe(operator.website);
    });
  });

  describe('when retrieving a single operator', () => {
    it('will return the operator if it exists', async () => {
      const data = parseOperatorJSON(TestData[0]);
      data.owner = owner.toEntity();
      await Operators.save(data);

      const operator = await service.getOperator(data.id);
      expect(operator).toBeDefined();
      expect(operator?.toJSON()).toMatchSnapshot();
    });

    it('will return undefined if the id cannot be found', async () => {
      await expect(
        service.getOperator('b32b8b8e-fbb1-4202-8c90-cf254118c5e1'),
      ).resolves.toBeUndefined();
    });
  });

  describe('when retrieving an operator by its slug', () => {
    it('will return the indicated operator', async () => {
      const data = parseOperatorJSON(TestData[0]);
      data.owner = owner.toEntity();
      await Operators.save(data);

      const operator = await service.getOperatorBySlug(data.slug.toUpperCase());
      expect(operator).toBeDefined();
      expect(operator?.toJSON()).toMatchSnapshot();
    });

    it('will return undefined if the operator does not exist', async () => {
      await expect(
        service.getOperatorBySlug('no-such-operator'),
      ).resolves.toBeUndefined();
    });
  });

  describe('when searching for operators', () => {
    let owners: UserEntity[];
    let searchData: OperatorEntity[];
    let activeOperatorsCount: number;

    beforeAll(() => {
      owners = TestOwners.slice(0, 80).map((user) => parseUserJSON(user));
      searchData = TestData.map((op, index) => {
        const data = parseOperatorJSON(op);
        data.owner = owners[index % owners.length];
        return data;
      });
      activeOperatorsCount = searchData.filter((op) => op.active).length;
    });

    beforeEach(async () => {
      await Users.save(owners);
      await Operators.save(searchData);
    });

    it('will perform a basic search with no parameters', async () => {
      const results = await service.searchOperators();
      expect(results.data).toHaveLength(50);
      expect(results.totalCount).toBe(activeOperatorsCount);
      expect(results.data[0]).toMatchSnapshot();
      expect(results.data.map((op) => op.name)).toMatchSnapshot();
    });

    it('will perform a search with pagination', async () => {
      const results = await service.searchOperators({ skip: 40, limit: 20 });
      expect(results.data).toHaveLength(20);
      expect(results.totalCount).toBe(activeOperatorsCount);
      expect(results.data.map((op) => op.name)).toMatchSnapshot();
    });

    it('will perform a search for dive operators near a given location', async () => {
      const results = await service.searchOperators({
        location: {
          lon: -55.6353,
          lat: -68.9415,
        },
        radius: 1500,
      });

      expect({
        length: results.data.length,
        totalCount: results.totalCount,
        operators: results.data.map((op) => op.name),
      }).toMatchSnapshot();
    });

    it('will perform a search for dive operators with a specific owner', async () => {
      const results = await service.searchOperators({
        owner: new User(Users, owners[2]),
      });

      expect(results.data).toHaveLength(3);
      expect(results.totalCount).toBe(3);
      expect(
        results.data.map((op) => ({
          name: op.name,
          owner: op.owner.username,
        })),
      ).toMatchSnapshot();
    });

    it('will perform a search with a query string', async () => {
      const results = await service.searchOperators({ query: 'urbanus' });
      expect({
        length: results.data.length,
        totalCount: results.totalCount,
        operators: results.data.map((op) => op.name),
      }).toMatchSnapshot();
    });

    it('will perform a search for dive operators including inactive operators', async () => {
      const results = await service.searchOperators({
        showInactive: true,
      });

      expect(results.data).toHaveLength(50);
      expect(results.totalCount).toBe(TestData.length);
      expect(
        results.data.map((op) => ({
          name: op.name,
          owner: op.owner.username,
          active: op.active,
        })),
      ).toMatchSnapshot();
    });

    it('will perform a search for dive operators with a specific verification status', async () => {
      const results = await service.searchOperators({
        verification: VerificationStatus.Pending,
      });
      const expectedCount = searchData.filter(
        (op) =>
          op.verificationStatus === VerificationStatus.Pending && op.active,
      ).length;

      expect(results.data).toHaveLength(expectedCount);
      expect(results.totalCount).toBe(expectedCount);
      expect(
        results.data.map((op) => ({
          name: op.name,
          owner: op.owner.username,
          verification: op.verificationStatus,
        })),
      ).toMatchSnapshot();
    });
  });
});
