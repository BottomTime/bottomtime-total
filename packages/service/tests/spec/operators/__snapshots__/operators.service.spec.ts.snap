// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OperatorService when retrieving a single operator will return the operator if it exists 1`] = `
{
  "active": true,
  "address": "1880 W Bridge Street Suite 633, South Felipestead, WY, 04394-9613",
  "averageRating": undefined,
  "banner": "https://loremflickr.com/1024/256?lock=7881950486331392",
  "createdAt": 1638440112327,
  "description": "Vomer carpo cogito pauci recusandae. Arto apto copiose animus. Xiphias civitas voluptatem ademptio tabesco viridis.
Compello atqui comprehendo theatrum caterva abstergo color thymbra vociferor benevolentia. Tutamen amplitudo curis atavus. Appono somniculosus delicate admoveo.",
  "email": "Jada_Okuneva@hotmail.com",
  "gps": {
    "lat": 6.9546,
    "lon": -52.6869,
  },
  "id": "850fe12a-1873-4a61-9e75-f793f1752688",
  "logo": "https://loremflickr.com/128/128?lock=2275161833209856",
  "name": "soft, ornery grape",
  "owner": {
    "accountTier": 0,
    "avatar": "https://example.com/avatar.png",
    "location": "Toronto, ON, Canada",
    "logBookSharing": "public",
    "memberSince": 1623294000000,
    "name": "Test User",
    "userId": "54ca5e54-de92-4e15-a523-4087c52b40eb",
    "username": "testuser",
  },
  "phone": "1-869-633-8778 x70097",
  "slug": "soft-ornery-grape",
  "socials": {
    "facebook": "Cheyenne.OReilly22",
    "instagram": "Nathaniel42",
    "tiktok": "Presley_Heaney91",
    "twitter": "Clemens30",
    "youtube": "Emmitt74",
  },
  "updatedAt": 1728680051726,
  "verificationMessage": undefined,
  "verificationStatus": "unverified",
  "website": "https://fake-endive.info/",
}
`;

exports[`OperatorService when retrieving an operator by its slug will return the indicated operator 1`] = `
{
  "active": true,
  "address": "1880 W Bridge Street Suite 633, South Felipestead, WY, 04394-9613",
  "averageRating": undefined,
  "banner": "https://loremflickr.com/1024/256?lock=7881950486331392",
  "createdAt": 1638440112327,
  "description": "Vomer carpo cogito pauci recusandae. Arto apto copiose animus. Xiphias civitas voluptatem ademptio tabesco viridis.
Compello atqui comprehendo theatrum caterva abstergo color thymbra vociferor benevolentia. Tutamen amplitudo curis atavus. Appono somniculosus delicate admoveo.",
  "email": "Jada_Okuneva@hotmail.com",
  "gps": {
    "lat": 6.9546,
    "lon": -52.6869,
  },
  "id": "850fe12a-1873-4a61-9e75-f793f1752688",
  "logo": "https://loremflickr.com/128/128?lock=2275161833209856",
  "name": "soft, ornery grape",
  "owner": {
    "accountTier": 0,
    "avatar": "https://example.com/avatar.png",
    "location": "Toronto, ON, Canada",
    "logBookSharing": "public",
    "memberSince": 1623294000000,
    "name": "Test User",
    "userId": "54ca5e54-de92-4e15-a523-4087c52b40eb",
    "username": "testuser",
  },
  "phone": "1-869-633-8778 x70097",
  "slug": "soft-ornery-grape",
  "socials": {
    "facebook": "Cheyenne.OReilly22",
    "instagram": "Nathaniel42",
    "tiktok": "Presley_Heaney91",
    "twitter": "Clemens30",
    "youtube": "Emmitt74",
  },
  "updatedAt": 1728680051726,
  "verificationMessage": undefined,
  "verificationStatus": "unverified",
  "website": "https://fake-endive.info/",
}
`;

exports[`OperatorService when retrieving multiple operators will retrieve several operators at once 1`] = `
[
  {
    "active": true,
    "address": "959 Alexys Burg Suite 955, North Bethesda, NE, 34813-8348",
    "averageRating": undefined,
    "banner": "https://loremflickr.com/1024/256?lock=4726526075469824",
    "createdAt": 1627998794619,
    "description": "Deduco denego audax tam. Quo conduco thermae urbs inventore vulgivagus auditor. Dicta cuppedia adsum creber turpis.
Cubicularis defleo creptio aggredior. Vaco casso quis audax strenuus. Similique nobis desolo bardus curso.",
    "email": "Ernestine51@yahoo.com",
    "gps": {
      "lat": 68.3899,
      "lon": -29.5619,
    },
    "id": "98ae6fdf-285c-4b4e-aade-79fd967f08bf",
    "logo": "https://picsum.photos/seed/lF5vXE6Fg/128/128",
    "name": "forked, striking developing",
    "owner": {
      "accountTier": 0,
      "avatar": "https://example.com/avatar.png",
      "location": "Toronto, ON, Canada",
      "logBookSharing": "public",
      "memberSince": 1623294000000,
      "name": "Test User",
      "userId": "54ca5e54-de92-4e15-a523-4087c52b40eb",
      "username": "testuser",
    },
    "phone": "(919) 571-9030 x9932",
    "slug": "forked-striking-developing",
    "socials": {
      "facebook": "Hettie_Mills",
      "instagram": "Kailee_Quitzon",
      "tiktok": "Bonnie_Becker44",
      "twitter": "Karen_Bogisich-Kunze",
      "youtube": "Maudie_Schmitt62",
    },
    "updatedAt": 1727764175796,
    "verificationMessage": undefined,
    "verificationStatus": "unverified",
    "website": "https://mild-egg.org/",
  },
  {
    "active": true,
    "address": "540 Sandringham Road Apt. 988, Creolachester, NE, 47119",
    "averageRating": undefined,
    "banner": "https://loremflickr.com/1024/256?lock=1946475686789120",
    "createdAt": 1687561706125,
    "description": "Tepidus caritas aiunt quis. Vociferor cotidie aliqua collum adnuo bonus admitto argentum defetiscor. Audax caput valens debilito coruscus adipiscor verumtamen.
Accusantium tollo casso excepturi verumtamen reprehenderit commodi abundans. Assumenda aeneus decens cogo auditor vulgaris excepturi spargo. Terra curo cicuta stillicidium acies antea.",
    "email": "Dillan14@yahoo.com",
    "gps": {
      "lat": 77.2672,
      "lon": -119.1238,
    },
    "id": "65704209-2883-4851-be84-78a6b95b9ad3",
    "logo": "https://loremflickr.com/128/128?lock=6087808152240128",
    "name": "miniature, unlawful hydrocarbon",
    "owner": {
      "accountTier": 0,
      "avatar": "https://example.com/avatar.png",
      "location": "Toronto, ON, Canada",
      "logBookSharing": "public",
      "memberSince": 1623294000000,
      "name": "Test User",
      "userId": "54ca5e54-de92-4e15-a523-4087c52b40eb",
      "username": "testuser",
    },
    "phone": "1-226-591-0255",
    "slug": "miniature-unlawful-hydrocarbon",
    "socials": {
      "facebook": "Isai.Stracke66",
      "instagram": "Adrienne.Stoltenberg53",
      "tiktok": "Johnson_Dicki64",
      "twitter": "Judd_Berge",
      "youtube": "Felipe3",
    },
    "updatedAt": 1720435396038,
    "verificationMessage": undefined,
    "verificationStatus": "unverified",
    "website": "https://exhausted-milepost.biz",
  },
  {
    "active": true,
    "address": "1880 W Bridge Street Suite 633, South Felipestead, WY, 04394-9613",
    "averageRating": undefined,
    "banner": "https://loremflickr.com/1024/256?lock=7881950486331392",
    "createdAt": 1638440112327,
    "description": "Vomer carpo cogito pauci recusandae. Arto apto copiose animus. Xiphias civitas voluptatem ademptio tabesco viridis.
Compello atqui comprehendo theatrum caterva abstergo color thymbra vociferor benevolentia. Tutamen amplitudo curis atavus. Appono somniculosus delicate admoveo.",
    "email": "Jada_Okuneva@hotmail.com",
    "gps": {
      "lat": 6.9546,
      "lon": -52.6869,
    },
    "id": "850fe12a-1873-4a61-9e75-f793f1752688",
    "logo": "https://loremflickr.com/128/128?lock=2275161833209856",
    "name": "soft, ornery grape",
    "owner": {
      "accountTier": 0,
      "avatar": "https://example.com/avatar.png",
      "location": "Toronto, ON, Canada",
      "logBookSharing": "public",
      "memberSince": 1623294000000,
      "name": "Test User",
      "userId": "54ca5e54-de92-4e15-a523-4087c52b40eb",
      "username": "testuser",
    },
    "phone": "1-869-633-8778 x70097",
    "slug": "soft-ornery-grape",
    "socials": {
      "facebook": "Cheyenne.OReilly22",
      "instagram": "Nathaniel42",
      "tiktok": "Presley_Heaney91",
      "twitter": "Clemens30",
      "youtube": "Emmitt74",
    },
    "updatedAt": 1728680051726,
    "verificationMessage": undefined,
    "verificationStatus": "unverified",
    "website": "https://fake-endive.info/",
  },
]
`;

exports[`OperatorService when searching for operators will perform a basic search with no parameters 1`] = `
{
  "active": true,
  "address": "47191 Astrid Lane Suite 737, Jarenmouth, MD, 18868",
  "averageRating": undefined,
  "banner": "https://loremflickr.com/1024/256?lock=1969575207895040",
  "createdAt": 1687732359551,
  "description": "Nobis aqua odio valens confugo spiculum viriliter. Explicabo torrens eos dolore socius natus facilis. Stipes audax spero uberrime apostolus.
Velit calamitas cogito decor defendo advenio allatus comes tenuis nulla. Praesentium tempore paulatim pel theatrum brevis a. Eos accommodo libero abscido solvo astrum solus.",
  "email": "Alverta.Torp@gmail.com",
  "gps": {
    "lat": 85.9108,
    "lon": 87.5066,
  },
  "id": "2e825650-4daf-4044-bc86-84513c4abc84",
  "logo": "https://loremflickr.com/128/128?lock=3880418432516096",
  "name": "absolute, testy prairie",
  "owner": {
    "accountTier": 0,
    "avatar": "https://cloudflare-ipfs.com/ipfs/Qmd3W5DuhgHirLHGVixi6V76LhCkZUz6pnFt5AJBiyvHye/avatar/1182.jpg",
    "location": "Evanston",
    "logBookSharing": "private",
    "memberSince": 1550440627852,
    "name": "Mrs. Israel Stracke",
    "userId": "9d72ecc0-b240-47c9-9d42-e6d196a3383f",
    "username": "Audreanne.Larkin79",
  },
  "phone": "270-756-3332 x7034",
  "slug": "absolute-testy-prairie",
  "socials": {
    "facebook": "Fernando.Armstrong56",
    "instagram": "Verona_Schultz",
    "tiktok": "Alessandra_Osinski",
    "twitter": "Andreanne47",
    "youtube": "Nettie_Parisian",
  },
  "updatedAt": 1722974726295,
  "verificationMessage": undefined,
  "verificationStatus": "unverified",
  "website": "https://golden-beggar.name",
}
`;

exports[`OperatorService when searching for operators will perform a basic search with no parameters 2`] = `
[
  "absolute, testy prairie",
  "acceptable, rotten efficacy",
  "adventurous, ripe flintlock",
  "annual, crowded editor",
  "another, upset harbour",
  "apt, dearest croissant",
  "apt, wicked aluminium",
  "arid, broken secretary",
  "athletic, poised cursor",
  "average, mad gondola",
  "awesome, pertinent spume",
  "baggy, measly tempo",
  "beneficial, knowledgeable gold",
  "bite-sized, pitiful entree",
  "bite-sized, unlined query",
  "black, considerate hypothermia",
  "bland, giant exhaust",
  "blue, evergreen fraudster",
  "bogus, thorny macaroni",
  "brief, interesting chairman",
  "bumpy, lean hospitalisation",
  "careful, worse baboon",
  "cautious, well-to-do supreme",
  "charming, linear analysis",
  "chief, smug blogger",
  "clear-cut, shady bend",
  "close, sizzling sugar",
  "clumsy, unripe maintainer",
  "composed, hungry potato",
  "considerate, organic spool",
  "content, formal clavicle",
  "conventional, empty wallaby",
  "cooked, well-to-do ghost",
  "crafty, roasted eligibility",
  "creative, slim escape",
  "creepy, precious butane",
  "creepy, unusual ephemera",
  "cute, delirious wound",
  "cute, emotional chord",
  "daring, delectable proliferation",
  "darling, insistent sparrow",
  "defensive, mealy choice",
  "delightful, tense pound",
  "descriptive, same infancy",
  "direct, lustrous slang",
  "dirty, flimsy buck",
  "distinct, fumbling marmalade",
  "dreary, similar mattock",
  "elaborate, deep cupcake",
  "elderly, frozen conspirator",
]
`;

exports[`OperatorService when searching for operators will perform a search for dive operators including inactive operators 1`] = `
[
  {
    "active": true,
    "name": "absolute, testy prairie",
    "owner": "Audreanne.Larkin79",
  },
  {
    "active": true,
    "name": "acceptable, rotten efficacy",
    "owner": "Jeffry_Kuphal38",
  },
  {
    "active": true,
    "name": "adventurous, ripe flintlock",
    "owner": "Steve54",
  },
  {
    "active": false,
    "name": "ajar, shady win",
    "owner": "Ken17",
  },
  {
    "active": true,
    "name": "annual, crowded editor",
    "owner": "Ken17",
  },
  {
    "active": true,
    "name": "another, upset harbour",
    "owner": "Malcolm89",
  },
  {
    "active": true,
    "name": "apt, dearest croissant",
    "owner": "Clarabelle_Funk69",
  },
  {
    "active": true,
    "name": "apt, wicked aluminium",
    "owner": "Jamison_Veum",
  },
  {
    "active": true,
    "name": "arid, broken secretary",
    "owner": "Jewel12",
  },
  {
    "active": true,
    "name": "athletic, poised cursor",
    "owner": "Lurline96",
  },
  {
    "active": true,
    "name": "average, mad gondola",
    "owner": "Logan41",
  },
  {
    "active": true,
    "name": "awesome, pertinent spume",
    "owner": "Brigitte.Kihn",
  },
  {
    "active": true,
    "name": "baggy, measly tempo",
    "owner": "Bryce_Thompson97",
  },
  {
    "active": true,
    "name": "beneficial, knowledgeable gold",
    "owner": "Trey.Langosh",
  },
  {
    "active": true,
    "name": "bite-sized, pitiful entree",
    "owner": "Lurline96",
  },
  {
    "active": true,
    "name": "bite-sized, unlined query",
    "owner": "Aiyana.McCullough",
  },
  {
    "active": true,
    "name": "black, considerate hypothermia",
    "owner": "Eleonore.Hilpert",
  },
  {
    "active": true,
    "name": "bland, giant exhaust",
    "owner": "Jaqueline19",
  },
  {
    "active": true,
    "name": "blue, evergreen fraudster",
    "owner": "Larue_Denesik32",
  },
  {
    "active": true,
    "name": "bogus, thorny macaroni",
    "owner": "Stephany76",
  },
  {
    "active": true,
    "name": "brief, interesting chairman",
    "owner": "Kendra.Haag",
  },
  {
    "active": true,
    "name": "bumpy, lean hospitalisation",
    "owner": "Donnell_Abbott",
  },
  {
    "active": true,
    "name": "careful, worse baboon",
    "owner": "Aiyana.McCullough",
  },
  {
    "active": true,
    "name": "cautious, well-to-do supreme",
    "owner": "Danielle56",
  },
  {
    "active": true,
    "name": "charming, linear analysis",
    "owner": "Okey84",
  },
  {
    "active": true,
    "name": "chief, smug blogger",
    "owner": "Jewel12",
  },
  {
    "active": true,
    "name": "clear-cut, shady bend",
    "owner": "Caesar71",
  },
  {
    "active": true,
    "name": "close, sizzling sugar",
    "owner": "Gerald79",
  },
  {
    "active": true,
    "name": "clumsy, unripe maintainer",
    "owner": "Ashlynn_Kreiger86",
  },
  {
    "active": true,
    "name": "composed, hungry potato",
    "owner": "Marie28",
  },
  {
    "active": false,
    "name": "confused, steel rocker",
    "owner": "Lorenz59",
  },
  {
    "active": true,
    "name": "considerate, organic spool",
    "owner": "Drew75",
  },
  {
    "active": true,
    "name": "content, formal clavicle",
    "owner": "Larue_Denesik32",
  },
  {
    "active": true,
    "name": "conventional, empty wallaby",
    "owner": "Delphia.Rohan22",
  },
  {
    "active": true,
    "name": "cooked, well-to-do ghost",
    "owner": "Liliane94",
  },
  {
    "active": true,
    "name": "crafty, roasted eligibility",
    "owner": "Amara.Cummings13",
  },
  {
    "active": true,
    "name": "creative, slim escape",
    "owner": "Ashly71",
  },
  {
    "active": true,
    "name": "creepy, precious butane",
    "owner": "Rene47",
  },
  {
    "active": true,
    "name": "creepy, unusual ephemera",
    "owner": "Aaron_Wehner28",
  },
  {
    "active": true,
    "name": "cute, delirious wound",
    "owner": "Dorothy.Hahn",
  },
  {
    "active": true,
    "name": "cute, emotional chord",
    "owner": "Jakob_Goyette",
  },
  {
    "active": true,
    "name": "daring, delectable proliferation",
    "owner": "Audreanne.Larkin79",
  },
  {
    "active": true,
    "name": "darling, insistent sparrow",
    "owner": "Sigurd_Reinger55",
  },
  {
    "active": false,
    "name": "decisive, shrill trail",
    "owner": "Jacey_Bernier",
  },
  {
    "active": true,
    "name": "defensive, mealy choice",
    "owner": "Rhianna.Terry",
  },
  {
    "active": true,
    "name": "delightful, tense pound",
    "owner": "Julia15",
  },
  {
    "active": true,
    "name": "descriptive, same infancy",
    "owner": "Jett_Block95",
  },
  {
    "active": false,
    "name": "descriptive, terrible today",
    "owner": "Alessia.Upton8",
  },
  {
    "active": true,
    "name": "direct, lustrous slang",
    "owner": "Felton36",
  },
  {
    "active": true,
    "name": "dirty, flimsy buck",
    "owner": "Donnell51",
  },
]
`;

exports[`OperatorService when searching for operators will perform a search for dive operators near a given location 1`] = `
{
  "length": 5,
  "operators": [
    "charming, linear analysis",
    "elaborate, deep cupcake",
    "impeccable, assured skeleton",
    "pungent, troubled artifact",
    "worst, unaware pride",
  ],
  "totalCount": 5,
}
`;

exports[`OperatorService when searching for operators will perform a search for dive operators with a specific owner 1`] = `
[
  {
    "name": "miniature, unlawful hydrocarbon",
    "owner": "Lucy55",
  },
  {
    "name": "noteworthy, annual pancake",
    "owner": "Lucy55",
  },
  {
    "name": "zigzag, frayed psychologist",
    "owner": "Lucy55",
  },
]
`;

exports[`OperatorService when searching for operators will perform a search for dive operators with a specific verification status 1`] = `
[
  {
    "name": "creative, slim escape",
    "owner": "Ashly71",
    "verification": "pending",
  },
  {
    "name": "heavenly, buttery distinction",
    "owner": "Jamey87",
    "verification": "pending",
  },
  {
    "name": "profitable, puzzling young",
    "owner": "Jefferey_Davis41",
    "verification": "pending",
  },
]
`;

exports[`OperatorService when searching for operators will perform a search with a query string 1`] = `
{
  "length": 10,
  "operators": [
    "acceptable, rotten efficacy",
    "baggy, measly tempo",
    "equatorial, wealthy electrocardiogram",
    "meek, sparkling tragedy",
    "messy, unimportant accord",
    "powerless, nocturnal movie",
    "quick-witted, excitable proponent",
    "sturdy, helpful hard-hat",
    "unhappy, slippery offering",
    "vigorous, average goddess",
  ],
  "totalCount": 10,
}
`;

exports[`OperatorService when searching for operators will perform a search with pagination 1`] = `
[
  "darling, insistent sparrow",
  "defensive, mealy choice",
  "delightful, tense pound",
  "descriptive, same infancy",
  "direct, lustrous slang",
  "dirty, flimsy buck",
  "distinct, fumbling marmalade",
  "dreary, similar mattock",
  "elaborate, deep cupcake",
  "elderly, frozen conspirator",
  "enchanted, stale neuropathologist",
  "equatorial, wealthy electrocardiogram",
  "every, unaware flugelhorn",
  "expensive, better pastry",
  "extra-small, accurate export",
  "faint, productive coaster",
  "fantastic, cool potato",
  "fantastic, excited criteria",
  "female, giant macrofauna",
  "few, joyous endoderm",
]
`;
